WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/eclipse/ui/handlers/StartupUIThread.java

```diff
@@ -56,19 +56,26 @@ public void run() {
 
 	/** Checks whether there is a registered WatchDog user */
 	private void checkWhetherToDisplayUserProjectRegistrationWizard() {
-		if (!WatchDogUtils.isEmpty(preferences.getUserid()))
+		ProjectPreferenceSetting projectSetting = preferences
+				.getOrCreateProjectSetting(workspaceName);
+		if (!WatchDogUtils.isEmpty(preferences.getUserid())
+				|| (projectSetting.startupQuestionAsked && !projectSetting.enableWatchdog))
 			return;
 
 		UserRegistrationWizardDialogHandler newUserWizardHandler = new UserRegistrationWizardDialogHandler();
 		try {
 			int statusCode = (int) newUserWizardHandler
 					.execute(new ExecutionEvent());
+			savePreferenceStoreIfNeeded();
 			if (statusCode == Window.CANCEL) {
 				if (MessageDialog.openQuestion(null, "WatchDog not active!",
 						WATCHDOG_INACTIVE_WARNING)) {
 					makeSilentRegistration();
 				} else {
 					userProjectRegistrationCancelled = true;
+					// user clearly does not WatchDog in this Project
```

Comment:
It means that a user cancelled Wizard, and also said that he doesn't want to use WatchDog anonymously (we don't register anything, we don't activate WD, we store in Preferences that this project shouldn't be asked about).